aspnetcore/migration/20_21.md

```diff
@@ -52,14 +51,19 @@ ASP.NET Core includes the following shared runtimes:
 * [Microsoft.AspNetCore.App](xref:fundamentals/metapackage-app)
 * [Microsoft.AspNetCore.All](xref:fundamentals/metapackage)
 
-Rules for projects targeting the shared runtime:
+The version specified by the package reference is the *minimum required* version. If a project references the 2.1.1 version of these packages, the app produced won't run on a machine that only has the 2.1.0 runtime installed.
+
+Known issues for projects targeting the shared runtime:
+
+* Using `Microsoft.AspNetCore.App` 2.1.0 with Entity Framework Core 2.1.1 causes NuGet restore failures due to package conflicts. The recommended solution is to upgrade `Microsoft.AspNetCore.App` to 2.1.1. For more information, see [Packages that share dependencies with Microsoft.AspNetCore.App cannot reference patch versions](https://github.com/aspnet/Universe/issues/1180).
+* All projects that must use `Microsoft.AspNetCore.All` or `Microsoft.AspNetCore.App` should add a package reference for the package in the project file, even if they contain a project reference to another project using `Microsoft.AspNetCore.All` or `Microsoft.AspNetCore.App`.
+
+  Example:
 
-* Projects referencing the `Microsoft.AspNetCore.All` or `Microsoft.AspNetCore.App` packages must set the project's SDK to `Microsoft.NET.Sdk.Web` at the top of the project file (`<Project Sdk="Microsoft.NET.Sdk.Web">`).
-* Projects referencing packages or projects that transitively reference `Microsoft.AspNetCore.All` or `Microsoft.AspNetCore.App`:
-  * Must set the project's SDK to `Microsoft.NET.Sdk.Web` at the top of the project file (`<Project Sdk="Microsoft.NET.Sdk.Web">`).
-  * Must reference the same shared runtime package. If LibraryA references `Microsoft.AspNetCore.App`, any projects referencing LibraryA must also reference `Microsoft.AspNetCore.App`.
-* Executable projects (projects that contain apps that are launched with `dotnet run` or apps that run and test projects for apps) must not specify a version for `Microsoft.AspNetCore.App`. The SDK specifies the version implicitly via `<PackageReference Include="Microsoft.AspNetCore.App" />`.
-* Referenced projects (projects that aren't the entry point and the project references `Microsoft.AspNetCore.All` or `Microsoft.AspNetCore.App`) must specify a package version.
+  - `MyApp` has a package reference to `Microsoft.AspNetCore.App`.
+  - `MyApp.Tests` has a project reference to `MyApp.csproj`.
+  
+  Add a add a package reference for `Microsoft.AspNetCore.App` to `MyApp.Tests`. For more information, see [Integration testing is hard to set up and may break on shared framework servicing](https://github.com/aspnet/Home/issues/3156).
```

Comment:
~~add a~~

Reply:
@natemcmaster See what I mean? :smile: That's ok ... more free :beer: for Scott.